src/app/components/dashboard/dashboard.service.js

```diff
@@ -214,6 +214,8 @@
                             $rootScope.properties.hostname = 'Hostname not available.';
                             $log.error('Error fetching hostname.');
                         });
+              
+					PMAPIService.getMetricsMetadata(data);
```

Comment:
No, actually now we need all the metadata to be stored somewhere, as we require the metadata to populate the list and provide details about the metric on hover via tooltip(mentioned in the stretch goals). So, I thought of storing the complete metadata as a `$rootScope` variable.